homeassistant/components/minecraft_server/quality_scale.yaml

```diff
@@ -0,0 +1,113 @@
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  entity-event-setup:
+    status: done
+    comment: Handled by coordinator.
+  entity-unique-id:
+    status: done
+    comment: Using confid entry ID as the dependency mcstatus doesn't provide a unique information.
+  has-entity-name: done
+  runtime-data: todo
+  test-before-configure: done
+  test-before-setup:
+    status: done
+    comment: |
+      Raising ConfigEntryNotReady, if either the initialization or
+      refresh of coordinator isn't successful.
+  unique-config-entry:
+    status: done
+    comment: |
+      As there is no unique information available from the dependency mcstatus,
+      the server address is used to identify that the same service is already configured.
+
+  # Silver
+  action-exceptions:
+    status: exempt
+    comment: Integration doesn't provide any service actions.
+  config-entry-unloading: done
+  docs-configuration-parameters:
+    status: exempt
+    comment: Integration doesn't support any configuration parameters.
+  docs-installation-parameters: done
+  entity-unavailable:
+    status: done
+    comment: Handled by coordinator.
+  integration-owner: done
+  log-when-unavailable:
+    status: done
+    comment: Handled by coordinator.
+  parallel-updates:
+    status: todo
+    comment: |
+      Although this is handled by the coordinator and no service actions are provided,
+      PARALLEL_UPDATES should still be set to 0 in binary_sensor and sensor according to the rule.
+  reauthentication-flow:
+    status: exempt
+    comment: No authentication is required for the integration.
+  test-coverage: done
+
+  # Gold
+  devices: done
+  diagnostics: done
+  discovery:
+    status: exempt
+    comment: No discovery possible.
+  discovery-update-info:
+    status: exempt
+    comment: |
+      No discovery possible. Users can use the (local or public) hostname instead of an IP address,
+      if static IP addresses cannot be configured.
+  docs-data-update: todo
+  docs-examples: todo
+  docs-known-limitations: todo
+  docs-supported-devices: done
+  docs-supported-functions: done
+  docs-troubleshooting: todo
+  docs-use-cases: todo
+  dynamic-devices:
+    status: exempt
+    comment: A minecraft server can only have one device.
+  entity-category: done
+  entity-device-class: done
+  entity-disabled-by-default: done
+  entity-translations: done
+  exception-translations: todo
+  icon-translations: done
+  reconfiguration-flow: todo
+  repair-issues:
+    status: exempt
+    comment: No repair use-cases for this integration.
+  stale-devices: todo
+
+  # Platinum
+  async-dependency:
+    status: done
+    comment: |
+      Lookup API of the dependency mcstatus for Bedrock Edition servers is not async,
+      but is non-blocking and therefore OK to be called. Refer to mcstatus FAQ
+      https://mcstatus.readthedocs.io/en/stable/pages/faq/#why-doesn-t-bedrockserver-have-an-async-lookup-method
```

Comment:
Wait, I am confused now, the bedrock server in the end does check the server status by connecting to the server right?

Reply:
No. From the linked FAQ: 

> With Java servers, to find the server, we sometimes end up performing an SRV DNS lookup. This means making a request to your DNS server and waiting for an answer, making that lookup a blocking operation (during which other things can be done).
> 
> [...]
> 
> However with Bedrock servers, no such lookups are required (Bedrock doesn’t support SRV records), and so there is no blocking I/O operation being made, that would justify having an async version.
> 
> In fact, all that the bedrock lookup does is parsing the host:port address, and obtaining the host and port parts out of it (with some error handling, and support for default ports).

Here is the link to the default lookup function, which is used in case of Bedrock Edition servers: https://github.com/py-mine/mcstatus/blob/fc3b96f614f44cbc74207022aa54c95c0bd7e44f/mcstatus/server.py#L47

Reply:
Yes I read that, but in the end you can get a status from thin air right? Or doesn't that come from the lookup method

Reply:
Ah but then I don't see a problem?

Reply:
That's also my opinion and why I set it to `done` 😄

Reply:
This conversation can then be set to resolved, right?